ext/intl/transliterator/transliterator_methods.c

```diff
@@ -424,7 +419,9 @@ PHP_FUNCTION( transliterator_transliterate )
 		RETVAL_FALSE;
 	}
 
-	zval_ptr_dtor( &tmp_object );
+	if (object != &tmp_object) {
+		zval_ptr_dtor( &tmp_object );
+	}
```

Comment:
I think you want to keep this as-is and use ZVAL_OBJ_COPY above. Right now you're leaking.